xsuite/src/world/csworld.ts

```diff
@@ -0,0 +1,272 @@
+    this.sysAcc = this.newContract(new Uint8Array(32).fill(255));
+  }
+
+  static new(options: CSWorldNewOptions) {
+    if (options.chainId !== undefined) {
+      throw new Error("chainId is not undefined.");
+    }
+    const { proxyUrl, gasPrice, explorerUrl, server, waitCompletedTimeout } =
+      options;
+    return new CSWorld({
+      proxy: new CSProxy({
+        proxyUrl,
+        explorerUrl,
+        autoGenerateBlocks: options.autoGenerateBlocks ?? true,
+        waitCompletedTimeout,
+      }),
+      gasPrice: gasPrice ?? 1000000000,
+      explorerUrl,
+      server,
+    });
+  }
+
+  static newDevnet(): World {
+    throw new Error("newDevnet is not implemented.");
+  }
+
+  static newTestnet(): World {
+    throw new Error("newTestnet is not implemented.");
+  }
+
+  static newMainnet(): World {
+    throw new Error("newMainnet is not implemented.");
+  }
+
+  static async start({
+    gasPrice,
+    explorerUrl,
+    autoGenerateBlocks,
+    configFilePath,
+    nodeOverrideFilePath,
+    debug,
+  }: {
+    gasPrice?: number;
+    explorerUrl?: string;
+    autoGenerateBlocks?: boolean;
+    configFilePath?: string;
+    nodeOverrideFilePath?: string;
+    debug?: boolean;
+  } = {}): Promise<CSWorld> {
+    const { server, proxyUrl } = await startCsproxyBin(
+      debug,
+      configFilePath,
+      nodeOverrideFilePath,
+    );
+    return CSWorld.new({
+      proxyUrl,
+      gasPrice,
+      explorerUrl,
+      server,
+      autoGenerateBlocks,
+    });
+  }
+
+  newWallet(addressOrSigner: AddressLike | Signer): CSWallet {
+    return new CSWallet({
+      signer: isAddressLike(addressOrSigner)
+        ? new DummySigner(addressOrSigner)
+        : addressOrSigner,
+      proxy: this.proxy,
+      chainId: this.chainId,
+      gasPrice: this.gasPrice,
+    });
+  }
+
+  newContract(address: string | Uint8Array): CSContract {
+    return new CSContract({
+      address,
+      proxy: this.proxy,
+    });
+  }
+
+  async createWallet({ address, ...params }: CSWorldCreateAccountParams = {}) {
+    address ??= generateWalletU8AAddress();
+    await setAccount(this.proxy, { address, ...params });
+    return this.newWallet(new DummySigner(address));
+  }
+
+  createContract(params?: CSWorldCreateAccountParams) {
+    return createContract(this.proxy, params);
+  }
+
+  setAccount(params: EncodableAccount) {
+    return setAccount(this.proxy, params);
+  }
+
+  generateBlocks(numBlocks: number) {
+    return this.proxy.generateBlocks(numBlocks);
+  }
+
+  generateBlock() {
+    return this.proxy.generateBlock();
+  }
+
+  generateBlocksUntilEpochReached(epoch: number) {
+    return this.proxy.generateBlocksUntilEpochReached(epoch);
+  }
+
+  getInitialWallets() {
+    return this.proxy.getInitialWallets();
+  }
+
+  terminate() {
+    if (!this.server) throw new Error("No server defined.");
+    killChildProcess(this.server);
+  }
+
+  getAccountNonce(address: AddressLike, shardId: number = undefined) {
```

Comment:
@lcswillems Not exactly sure how you want to handle the forced shard get account, is this right?

Reply:
Yes, I think so.

But it should be implemented at the Proxy level. Will take care of it.